### DISCUSSION TITLE
How to Avoid Filenames Being Overwritten

### DISCUSSION THREAD

**Discussion by KarstenKohout** [beetbox/beets#5351] [Q&A] [answered]
Hey all!

I am using beets to write metadata to existing library of music.  I just want it to write tags and not adjust the actual filenames.  This library is being sourced by a program that relies on the file name and the file location - hence the desire to avoid changing the file name.

Is there a way to avoid filenames being overwritten?

Thank you!

**Comment by Serene-Arc** [comment, marked as answer]
If you use the following, it shouldn't change the files' location or names, just the metadata, but definitely test on a small subset first.

```yaml
import:
	copy: no
	move: no
	write: yes
```

**Reply by KarstenKohout** [reply]
Oh great! My interpretation of the write command is that it would change the filename and location - I'll have to give this a shot.

Working over the weekend but will test on a sample when I get a moment.

Thank you so much!

**Reply by Serene-Arc** [reply]
The write command is specifically for metadata. I hope it works for you!

**Reply by KarstenKohout** [reply]
Just realized I never closed the loop here.  This solved my issue!  Apologies that this is likely a rookie mistake - I greatly appreciate your response!

**Comment by amogus07** [comment]
Which program is it, if I may ask? Maybe beets could be configured to use a folder/filename strategy that's compatible with the program.

**Reply by KarstenKohout** [reply]
Hello!

I am a mobile DJ and am using beets to refine my Serato DJ library.  I've thought about creating a plugin that contains configurations that are ideal for this application.  I'm just starting my CS journey now via a career change certificate program.  This is likely a ways off, but maybe I'll get it done someday!

I think it would be difficult to config a fresh instance of beets to play well with Serato because many DJ's already have existing libraries of music that are sourced from multiple locations.  There is also the reliability concern as there is no room for error during an event for Serato to unexpectedly not be able to locate and playback a track.

Currently, I'm copying the files I want to edit from my Windows machine onto a NAS that is accessible by the Ubuntu VM which runs beets, using beets there, then copying the files back to the NAS which is accessible by the Windows machine.  Not an incredibly graceful process, but I've created a SOP doc that has all the relevant command prompts to do it without UI.

Thanks!

**Reply by amogus07** [reply]
Yeah, not having beets handle folder/filenames is probably the best thing to do here. But you could consider having beets manage tags. What's keeping you from running it directly on your library with move and copy disabled?

**Reply by KarstenKohout** [reply]
That's actually what I'm doing (having beets manage tags)!  This was also the reason for my original question.

I am effectively running it directly on my library, I'm just doing it on copied versions in increments on a VM so that I have the ability to verify that Serato can still locate the files before fully replacing the files with the versions that have updated tags.

**Reply by amogus07** [reply]
Try running `beet update -p` on your Serato library (with the same config except for `directory`) and make sure the output doesn't indicate that anything would be deleted from beets' database! Otherwise you'd have to re-import your music all over again or find another solution.

**Reply by amogus07** [reply]
oh, forgot to mention, also point it to a copy of your current database
